src/modules/aws_ec2/mappers/general_purpose_volume.ts

```diff
@@ -246,10 +197,9 @@ export class GeneralPurposeVolumeMapper extends MapperBase<GeneralPurposeVolume>
     create: async (es: GeneralPurposeVolume[], ctx: Context) => {
       const out = [];
       for (const e of es) {
+        if (e.isRootDevice) continue; // cannot create root volumes, skip
+        if (e.volumeId) continue; // cannot create a volume that is already created, skip
```

Comment:
If someone mutates the definition of a volume that's already created, this will cause it to always get reverted during `commit`. Is that the desired behavior?

Reply:
i noticied that behaviour, and what i am doing in case of the replace case in update is something like ```        } else {
          // Replace, but first remove volume id
          e.volumeId = undefined;
          const newVolume = await this.module.generalPurposeVolume.cloud.create(e, ctx);
          await this.module.generalPurposeVolume.cloud.delete(cloudRecord, ctx);
          out.push(newVolume);
        }
```
so i let the volume be created again

Reply:
hmm but if the `volumeId` was edited that should not trigger a replacement but restore the record, right?